pkg/host/machine_info_linux.go

```diff
@@ -145,11 +147,53 @@ func getModulesInfo() ([]KernelModule, error) {
 func getGlobsInfo(globs map[string]bool) (map[string][]string, error) {
 	files := make(map[string][]string, len(globs))
 	for glob := range globs {
-		matches, err := filepath.Glob(glob)
-		if err != nil {
-			return nil, err
+		var (
+			addglobs []string
+			subglobs []string
+			matches  []string
+		)
+		tokens := strings.Split(glob, ":")
+		for _, tok := range tokens {
+			if strings.HasPrefix(tok, "-") {
+				subglobs = append(subglobs, tok[1:])
+			} else {
+				addglobs = append(addglobs, tok)
+			}
+		}
+		for _, g := range addglobs {
+			m, err := filepath.Glob(g)
+			if err != nil {
+				return nil, err
+			}
+			matches = append(matches, m...)
 		}
-		files[glob] = matches
+		files[glob] = excludeGlobs(removeDupValues(matches), subglobs)
 	}
 	return files, nil
 }
+
+func excludeGlobs(items, excludes []string) []string {
+	var results []string
+	for _, exclude := range excludes {
+		g := glob.MustCompile(exclude)
+		for _, item := range items {
+			if !g.Match(item) {
+				results = append(results, item)
+			}
+		}
+		items = results
+	}
+	return results
+}
+
+func removeDupValues(slice []string) []string {
+	keys := make(map[string]bool)
+	list := []string{}
+	for _, entry := range slice {
+		if _, ok := keys[entry]; !ok {
```

Comment:
It's more idiomatic and shorter to use just `if !keys[entry] {` for bool-valued maps. The missing keys will return `false`.